runtime/caml/bigarray.h

```diff
@@ -34,6 +34,7 @@ typedef unsigned short caml_ba_uint16;
 #define CAML_BA_MAX_NUM_DIMS 16
 
 enum caml_ba_kind {
+  CAML_BA_FLOAT16,             /* Half-precision floats */
```

Comment:
Moving this new constructor at the end would probably help with backward compatibility.
In its current form, marshaling a bigarray gives a different result from previous ocaml version because all constructors have been shifted by 1.

Reply:
@Octachron, @gasche, should I open an issue to track this properly ?

Reply:
My understanding is that @xavierleroy and @damiendoligez are on it (so: no need for a separate issue), but this is tricky because it requires an acrobatic bootstrap. Thanks for the report, this is indeed a good catch.

Reply:
Fyi, it was spotted while adapting jsoo to trunk

Reply:
I like this suggestion.  Implemented in #12357 .